xclim/sdba/adjustment.py

```diff
@@ -121,13 +125,48 @@ def _check_inputs(cls, *inputs, group):
             )
 
     @classmethod
-    def _harmonize_units(cls, *inputs, target: str | None = None):
+    def _harmonize_units(cls, *inputs, target: dict[str] | str | None = None):
         """Convert all inputs to the same units.
 
         If the target unit is not given, the units of the first input are used.
 
         Returns the converted inputs and the target units.
         """
+
+        def _harmonize_units_multivariate(
+            *inputs, dim, target: dict[str] | None = None
+        ):
+            def _convert_units_to(inda, dim, target):
+                for iv, v in enumerate(inda[dim].values):
+                    inda.attrs["units"], inda.attrs["standard_name"] = (
+                        inda[dim].attrs[a][iv] for a in ["_units", "_standard_name"]
+                    )
+                    inda[{dim: iv}] = convert_units_to(
+                        inda[{dim: iv}], target[v], context="infer"
+                    )
+                    inda[dim].attrs["_units"][iv] = target[v]
+                inda.attrs["units"], inda.attrs["standard_name"] = "", ""
+                return inda
+
+            if target is None:
+                target = {
+                    v: inputs[0][dim].attrs["_units"][iv]
+                    for iv, v in enumerate(inputs[0][dim].values)
+                }
+            return (
```

Comment:
If it is not too messy, I would try to add a fast path if all inputs already have the same units. The univariate function relies on the fact that `convert_units_to` already does that, but here we are provoking operations even if the units are the same.

Also, does this work smoothly with dask ?

Reply:
It's only attributing some dict values in the end, so there is not much gain with this fast pass, but hopefull the fast path is not too messy